### PR TITLE
core(full-page-screenshot): drop max datauri size constraints

### PR DESCRIPTION
**Pull request by paulirish**
(this PR is building off https://github.com/GoogleChrome/lighthouse/pull/11688 )

back here i discovered this data uri limit that was in the implementation wasn't a real limit. **https://github.com/GoogleChrome/lighthouse/pull/11402#issuecomment-692287106**  patrick confirmed.

I made another test here:

image

The chromium bug regarding that limit: https://bugs.chromium.org/p/chromium/issues/detail?id=69227  Yah as patrick said, it's likely this limit is about the data uri that you can **navigate** to in chrome. I definitely can't nav to a 2.5MB data uri. But it's totally fine as an ``

### CODE REVIEW

File: lighthouse-core/gather/gatherers/full-page-screenshot.js [hunk 1]
```diff
@@ -72,20 +69,7 @@ class FullPageScreenshot extends Gatherer {
...
-      // So just cutting down the height a bit usually fixes the issue.
-      screenshot = await this._takeScreenshot(passContext, 5000);
-      if (screenshot.data.length > MAX_DATA_URL_SIZE) {
-        passContext.LighthouseRunWarnings.push(
-          'Full page screenshot is too big–report won\'t show element screenshots.');
-        return null;
-      }
-    }
-
+    let screenshot = await this._takeScreenshot(passContext);
```
Comment (adamraine) [inline lighthouse-core/gather/gatherers/full-page-screenshot.js:72]: nit: Can we remove the `afterPass_` function and call `_takeScreenshot` directly?

File: lighthouse-core/gather/gatherers/full-page-screenshot.js [hunk 2]
```diff
@@ -119,8 +103,6 @@ class FullPageScreenshot extends Gatherer {
           return {
             width: document.documentElement.clientWidth,
             height: document.documentElement.clientHeight,
-            screenWidth: window.screen.width,
```
Comment (adamraine) [inline lighthouse-core/gather/gatherers/full-page-screenshot.js:122]: Was this necessary before this patch? Looks like `screenWidth` and `screenHeight` are not change in this gatherer.

Comment (paulirish) [reply lighthouse-core/gather/gatherers/full-page-screenshot.js]: fair. i meant to include it in #11688 instead. just moved it there.

**Review by adamraine — APPROVED** [review]
LGTM

**Comment by connorjclark** [comment]
Did you test in other browsers?

**Comment by paulirish** [comment]
> Did you test in other browsers?

tested with a 14.9MB data uri string. works great in chrome/saf/ff.

 
> Also, this isn't just preventing a possibly faulty image, it also attempts to put a cap on how much data a screenshot will be (the soon-to-be largest part of the LHR). For that purpose, maybe we ought to just lessen the quality instead of capping the height?

kinda frustrating....

so i feel confident i've put the max_datauri_size issue to bed. the comments/code were wrong and i've proven that.

as for LHR data size.. agree the FPS is big. but, we don't have any criteria to determine an LHR is _too_ big. So I don't see a great path forward in picking a threshold to retake a smaller screenshot.

**Comment by connorjclark** [comment]
> so i feel confident i've put the max_datauri_size issue to bed. the comments/code were wrong and i've proven that.

Yea, we're on the same page here. This change LGTM

> as for LHR data size.. agree the FPS is big. but, we don't have any criteria to determine an LHR is too big. So I don't see a great path forward in picking a threshold to retake a smaller screenshot.

Just pointing out the secondary effect of this change. If we don't want to address it now, at least we might recall this is a way to shrink an LHR in the future if a specific need arises.

**Pull request merged**